cpp/tests/test_abm_model.cpp

```diff
@@ -798,3 +809,64 @@ TEST(TestModel, mobilityTripWithAppliedNPIs)
     // The person does not want to isolate when the test is positive
     EXPECT_FALSE(p_no_isolation.is_in_quarantine(t, model.parameters));
 }
+
+TEST(TestModel, personCanDieInHospital)
+{
+    using testing::Return;
+
+    auto t     = mio::abm::TimePoint(0);
+    auto dt    = mio::abm::days(1);
+    auto model = mio::abm::Model(num_age_groups);
+
+    // Time to go from infected with symptoms to severe infection is dt.
+    model.parameters.get<mio::abm::InfectedSymptomsToSevere>()[{mio::abm::VirusVariant::Wildtype, age_group_60_to_79}] = 1;
+    // Time to go from severe infection to dead is dt.
+    model.parameters.get<mio::abm::SevereToDead>()[{mio::abm::VirusVariant::Wildtype, age_group_60_to_79}] = 1;
+
+    auto home_id     = model.add_location(mio::abm::LocationType::Home);
+    auto work_id     = model.add_location(mio::abm::LocationType::Work);
+    auto icu_id      = model.add_location(mio::abm::LocationType::ICU);
+    auto hospital_id = model.add_location(mio::abm::LocationType::Hospital);
+
+    // Make sure all persons will be InfectedSevere before Dead.
+    ScopedMockDistribution<testing::StrictMock<MockDistribution<mio::UniformDistribution<double>>>>
+        mock_uniform_dist;
+    EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)
+        .WillRepeatedly(testing::Return(0.3));
+
+    // Create a person that has severe infection and move to hospital at time t. The person is dead at time t + dt
+    add_test_person(model, home_id, age_group_60_to_79, mio::abm::InfectionState::Dead, t + dt);
+
```

Comment:
Here, perhaps we can check that the infection course will be Severe->Dead, so that there is no Critical step in between?

Reply:
Yes, I added some parameter settings from Infection Severe -> Critical -> Dead. A test for transition Severe->Dead is also added.